### ISSUE TITLE
message "semble manquant, il a été désactivé."

### ISSUE THREAD

**Issue by Krankhore** [Jeremie-C/plugin-MerossIOT#11]
J'ai fait la mise à jour du plugin avec la version du 2020-05-12 01:06:43 il y a quelques jours.
J'ai des prises MSS310 et j'ai de temps en temps l'apparition dans les massages de jeedom  de **"[équipement] semble manquant, il a été désactivé."** 

Ce n'est pas toujours la même prise et il n'y en qu'une à chaque fois.
Il n'y a pas de messages dans la logs de Merossiot

Pour autant, la prise n'est pas désactivé, les informations de consommation, tension intensité .. continuent d'être collecté. Tout semble normal.
C'est un message que j'avais déjà eu dans des versions précédentes du plugins et qui avait disparu depuis plusieurs versions.

Sinon, avec un recul de quelques jours, cette version semble plus stable pour la connexion à Meross. J'ai encore la tension qui est collecté à 0 de temps de en temps mais cela ne dure pas. Alors qu'avant la collecte ne refonctionnait normalement qu'après que j'ai relancé le démon et j'avais même eu des cas, ou il fallait que je redémarre complètement jeedom pour reprendre un fonctionnement normal (le symptôme était une collecte de consommation, tension, intensite normale , la collecte suivant à zero, puis à nouveau consommation,tension,intensite  normale etc..).

**Comment by Krankhore** [comment]
Info complémentaire, dans la page santé, l'adresse IP de l'équipement concerné par le message  ne figure plus mais la date de dernière communication est bien à jour. Pour les autres équipements l'adresse IP figure.
Si je fais synchroniser ( que ce soit au niveau de l'équipement ou le synchroniser général du plugin), l'adresse ip revient.

**Comment by Jeremie-C** [comment]
Bonjour,

il s'agit d'un fonctionnement "normal".
Lors de la synchro automatique, il se peut qu'un équipement soit renvoyé hors-ligne par le cloud meross et donc le plugin en informe Jeedom.
Lors de la synchro suivante, il est renvoyé en ligne et donc se réactive automatiquement sans message d'info du plugin.

L'ip ne se met pas à jour lors d'un retour en ligne, je travail dessus.

Je travail aussi sur une autre façon de gérer ces retours hors-ligne / en ligne des appareils afin d'éviter ces messages.

**Comment by Krankhore** [comment]
Oui, je comprends; mais cela reste un peu bizarre. Car cela avait disparu suite à une mise à jour du plugin et depuis la mise à jour du 12 mai, j'ai de nouveau régulièrement le message.

Les 7 prises sont sur le même réseau wifi, et le message n'apparait que pour une seule et ce n'est jamais pour la même prise.

Cela arrive aussi sur des prises actives en permanence avec de la consommation en continue et qui donc devrait remonter des infos en continu vers le cloud.

La synchro automatique est faite à quelle heure ? par exemple aujourd'hui le message est daté de 7h25.
Et apparemment, dans les versions précédentes, le démon était relancé toutes les nuits à 0h00 et ce n'est plus le cas, c'est bien ça ?

**Comment by Krankhore** [comment]
Aujourd'hui, j'ai eu deux équipements concernés à 12h01.
J'avais monté une alerte lorsque la tension remontée par le plugin est à zéro. Et c'est exactement l'heure a laquelle j'ai eu le message ou j'ai également eu cette alerte tension à zero.

Depuis cette version du plugin , cette alerte ne dure pas puisque la reconnexion au cloud se fait en automatique ( avant je devais relancer le démon voir relancer jeedom, c'est super cette récupération automatique). 
Est ce que le message "semble manquant"  ne pourrait pas être lié à la perte de la connexion ou de l'authentification vis à vis du cloud mal géré dans le sens ou tu traiterais des infos sans t'être encore aperçu que l’authentification était perdu. 
Car la situation de tension à zéro n'est normalement pas possible. Car si la prise remonté des infos vers le cloud, c'est qu"elle est branché. Et si elle est branché , la tension en non nulle !
Et le message "semble manquant" dans ce cas me parait est du même genre , à savoir info tronqué ou aberrante sur une conexion qui ne doit plus être valide.

**Comment by MrPatate87** [comment]
Bonjour j"ai rencontré le même soucis aujourd'hui. De plus ne tiens a remontrer l'issue qui faisait part de la non remontée d'info après une certain temps. En effet le plugin marche bien l'espace d'une journée mais par la suite il n'y à plus de remontée d'info (quelqu'elle soit). Pour qu cela fonctionne à nouveau il faut alors relancer l'installation des dépendances.

**Comment by Jeremie-C** [comment]
@Krankhore 
C'est mon plugin qui met la tension à 0 si l’équipement est détecté hors ligne. C'est en cours de correction.
Ce n'est pas lié à une déconnexion au cloud car le parcours des équipements test la connexion avant de récupérer le statut des équipements.

@MrPatate87 
Selon moi, vous n'êtes pas à jour du plugin ou lors de la mise à jour les dépendances ne se sont pas mises à jour.
Si ce n'est pas le cas, merci de créer une autre "issue" avec vos logs afin que je puisse traiter au mieux car ce n'est pas le même problème.
